tests/Unit/Utilities/Test_TMPLDocumentation.cpp

```diff
@@ -0,0 +1,1467 @@
+/// [math_arithmetic]
+
+/// [math_bitwise]
+assert_same<tmpl::complement<tmpl::uint8_t<0b10001111>>::type,
+                             tmpl::uint8_t<0b01110000>>();
+assert_same<tmpl::bitand_<tmpl::uint8_t<0b00111011>,
+                          tmpl::uint8_t<0b01010110>>::type,
+                          tmpl::uint8_t<0b00010010>>();
+assert_same<tmpl::bitor_<tmpl::uint8_t<0b01100011>,
+                         tmpl::uint8_t<0b10100111>>::type,
+                         tmpl::uint8_t<0b11100111>>();
+assert_same<tmpl::bitxor_<tmpl::uint8_t<0b11000011>,
+                          tmpl::uint8_t<0b00000110>>::type,
+                          tmpl::uint8_t<0b11000101>>();
+assert_same<tmpl::shift_left<tmpl::uint8_t<0b10001110>, tmpl::size_t<3>>::type,
+                             tmpl::uint8_t<0b01110000>>();
+assert_same<tmpl::shift_right<tmpl::uint8_t<0b10110011>, tmpl::size_t<4>>::type,
+                              tmpl::uint8_t<0b00001011>>();
+/// [math_bitwise]
+
+/// [math_comparison]
+assert_same<tmpl::equal_to<tmpl::size_t<1>, tmpl::size_t<2>>::type,
+            tmpl::false_type>();
+assert_same<tmpl::not_equal_to<tmpl::size_t<1>, tmpl::size_t<2>>::type,
+            tmpl::true_type>();
+assert_same<tmpl::greater<tmpl::size_t<1>, tmpl::size_t<2>>::type,
+            tmpl::false_type>();
+assert_same<tmpl::greater_equal<tmpl::size_t<1>, tmpl::size_t<2>>::type,
+            tmpl::false_type>();
+assert_same<tmpl::less<tmpl::size_t<1>, tmpl::size_t<2>>::type,
+            tmpl::true_type>();
+assert_same<tmpl::less_equal<tmpl::size_t<1>, tmpl::size_t<2>>::type,
+            tmpl::true_type>();
+/// [math_comparison]
+
+/// [math_logical]
+assert_same<tmpl::and_<std::true_type, std::false_type>::type,
+            tmpl::false_type>();
+assert_same<tmpl::or_<std::true_type, std::false_type>::type,
+            tmpl::true_type>();
+assert_same<tmpl::xor_<std::true_type, std::false_type>::type,
+            tmpl::true_type>();
+assert_same<tmpl::not_<std::true_type>::type, tmpl::false_type>();
+/// [math_logical]
+
+/// [tmpl::identity]
+assert_same<tmpl::identity<tmpl::size_t<10>>::type, tmpl::size_t<10>>();
+/// [tmpl::identity]
+
+/// [tmpl::max]
+assert_same<tmpl::max<tmpl::size_t<10>, tmpl::int32_t<3>>::type,
+            tmpl::size_t<10>>();
+/// [tmpl::max]
+
+/// [tmpl::min]
+assert_same<tmpl::min<tmpl::size_t<10>, tmpl::size_t<3>>::type,
+            tmpl::size_t<3>>();
+/// [tmpl::min]
+
+/// [tmpl::next]
+assert_same<tmpl::next<tmpl::size_t<10>>::type, tmpl::size_t<11>>();
+/// [tmpl::next]
+
+/// [tmpl::prev]
+assert_same<tmpl::prev<tmpl::size_t<10>>::type, tmpl::size_t<9>>();
+/// [tmpl::prev]
+}
+}  // namespace math_functions
+
+namespace miscellaneous {
+/// [tmpl::has_type:pack_expansion]
+template <typename... T>
+bool check_sizes(const T&... containers,
+                 const typename tmpl::has_type<T, size_t>::type... sizes) {
+  return (... && (containers.size() == sizes));
```

Comment:
Yay fold expressions :D Minor comment, could `and` be used instead of `&&` and would that be reasonable to do?

Reply:
I didn't think to try.  It probably works, since `and` is just token replacement.